### PR TITLE
Model Layer Testing POC

### PR DESCRIPTION
**Pull request by akhilg-nv**
## Test Cost Comparison

sanity test with TP=8

`time python -m unittest registered.modeling.test_modeling_deepseek_v3 -k sanity`

real    1m1.416s
user    6m30.110s
sys     1m36.591s

`time python -m unittest registered.8-gpu-models.test_deepseek_v3_basic -v`

Initial Run:
real    22m18.176s
user    1m40.294s
sys     0m31.521s

Second run (model loads much faster, DeepGEMM JIT pre-compilation already complete):
real    2m26.941s
user    1m40.806s
sys     0m24.931s

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: test/registered/modeling/test_modeling_deepseek_v3.py [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+import torch
+from parameterized import parameterized_class
+from transformers import AutoConfig, AutoModelForCausalLM, AutoTokenizer
+
+from sglang.srt.environ import envs
+from sglang.test.server_fixtures.default_fixture import DefaultServerBase
+
+MODEL = "deepseek-ai/DeepSeek-V3"
+MODEL_OVERRIDE_ARGS = {
+    "num_hidden_layers": 2,
```
Comment (nvpohanh) [inline test/registered/modeling/test_modeling_deepseek_v3.py]: Can we run with 4 layers + `first_k_dense_replace` set to 2? That means, 2 MLP layers + 2 MoE layers so that we can cover the communications across different layer boundaries (MLP-MLP, MLP-MoE, MoE-MoE)? Thanks

File: test/registered/modeling/test_modeling_deepseek_v3.py [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+    "first_k_dense_replace": 0,
+    "n_routed_experts": 8,
```
Comment (nvpohanh) [inline test/registered/modeling/test_modeling_deepseek_v3.py]: Let's keep `n_routed_experts` `num_experts_per_tok` `n_group` `topk_group` as the original because lots of DSV3/DSR1 code paths have hardcoded check on these parameters.

File: test/registered/modeling/test_modeling_deepseek_v3.py [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+        args.extend(["--attention-backend", cfg["attention_backend"]])
+
+    # Chunked prefill
+    if not cfg["chunked_prefill"]:
+        args.extend(["--chunked-prefill-size", "-1"])
+
+    # CUDA graph
+    if not cfg["cuda_graph"]:
+        args.append("--disable-cuda-graph")
+
```
Comment (nvpohanh) [inline test/registered/modeling/test_modeling_deepseek_v3.py:136]: Let's also add:
- `--kv-cache-dtype fp8_e4m3` or no flag
- `--moe-dense-tp-size 1` or no flag

File: test/registered/modeling/test_modeling_deepseek_v3.py [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+        "--trust-remote-code",
+        "--mem-fraction-static",
+        "0.5",
+    ]
+
+    if dummy_weights:
+        args.extend(["--load-format", "dummy"])
+
+    # Precision / quantization
+    precision = cfg["precision"]
```
Comment (nvpohanh) [inline test/registered/modeling/test_modeling_deepseek_v3.py]: I think precision and quantization should be two separated fields because we could have different combinations like bf16+fp8 and fp16+fp8, etc. Usually, "precision" defines what the high-precision dtype part (like linear op output) should be and "quantization" defines what the low-precision dtype part (like linear op input) should be

File: test/registered/modeling/test_modeling_deepseek_v3.py [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+                input_ids[0, i]
+            ].item()
+            diff = abs(sg_logprob - hf_token_logprob)
+            self.assertLess(
+                diff,
+                PREFILL_TOLERANCE,
+                f"Prefill logprob mismatch at position {i}: "
+                f"SGLang={sg_logprob:.6f}, HF={hf_token_logprob:.6f}, diff={diff:.6f}",
+            )
+
```
Comment (nvpohanh) [inline test/registered/modeling/test_modeling_deepseek_v3.py:450]: Can we run one prefill step and one decode step? You can refer to the TRT-LLM test as an example: https://github.com/NVIDIA/TensorRT-LLM/blob/main/tests/unittest/_torch/modeling/test_modeling_llama.py#L207-L379
